diff --git a/advpoll.install b/advpoll.install
index 94e7804..c9d3bb6 100644
--- a/advpoll.install
+++ b/advpoll.install
@@ -104,7 +104,9 @@ function advpoll_uninstall() {
 
   if ($nid) {
 
-    db_delete('votingapi_vote')->condition('entity_id', $nid, 'IN')->execute();
+    db_delete('votingapi_vote')->condition('entity_id', $nid, 'IN')
+      ->condition('entity_type', 'advpoll')
+      ->execute();
 
     // Delete all the nodes at once
     // http://api.drupal.org/api/function/node_delete_multiple/7
diff --git a/advpoll.module b/advpoll.module
index b4f9592..d4c8d83 100644
--- a/advpoll.module
+++ b/advpoll.module
@@ -323,7 +323,9 @@ function advpoll_node_presave($node) {
 
 function advpoll_node_delete($node) {
   // Clear node's votes from the voting api
-  db_delete('votingapi_vote')->condition('entity_id', $node->nid)->execute();
+  db_delete('votingapi_vote')->condition('entity_id', $node->nid)
+    ->condition('entity_type', 'advpoll')    
+    ->execute();
 }
 
 /*
diff --git a/advpoll_ranking/advpoll_ranking.install b/advpoll_ranking/advpoll_ranking.install
index 20fe3e3..5832cf6 100644
--- a/advpoll_ranking/advpoll_ranking.install
+++ b/advpoll_ranking/advpoll_ranking.install
@@ -25,7 +25,9 @@ function advpoll_ranking_uninstall() {
   }
 
   if ($nid) {
-    db_delete('votingapi_vote')->condition('entity_id', $nid, 'IN')->execute();
+    db_delete('votingapi_vote')->condition('entity_id', $nid, 'IN')
+      ->condition('entity_type', 'advpoll')      
+      ->execute();
     // Delete all the nodes at once
     // http://api.drupal.org/api/function/node_delete_multiple/7
     node_delete_multiple($nid);
diff --git a/includes/advpoll.pages.inc b/includes/advpoll.pages.inc
index 3ebaafb..4a1763a 100644
--- a/includes/advpoll.pages.inc
+++ b/includes/advpoll.pages.inc
@@ -367,7 +367,9 @@ function advpoll_clear_votes_confirm_form($form, &$form_state, $nid, $title) {
 
 function advpoll_clear_votes_confirm_form_submit($form, &$form_state) {
   $nid = (int) check_plain(arg(1));
-  db_delete('votingapi_vote')->condition('entity_id', $nid)->execute();
+  db_delete('votingapi_vote')->condition('entity_id', $nid)
+    ->condition('entity_type', 'advpoll')    
+    ->execute();
   drupal_goto('node/' . $nid);
 }
 
